asf_heat_pump_suitability/pipeline/reweight_epc/evaluate_reweighting.py

```diff
@@ -0,0 +1,273 @@
+"""
+Evaluate the effect of reweighting by calculating the errors before and after reweighting.
+
+Errors are calculated between the sample (unweighted and weighted) and target proportions of properties for each feature per LSOA.
+
+Run with:
+
+python asf_heat_pump_suitability/pipeline/reweight_epc/evaluate_reweighting.py
+	--reweighted_dir "s3://asf-heat-pump-suitability/outputs/2023_Q2_EPC_enhanced_weights.parquet"
+	--sample
+
+[remove the --sample argument to run on full dataset]
+
+"""
+
+import polars as pl
+import polars.selectors as cs
+from tqdm import tqdm
+
+from argparse import ArgumentParser
+
+from asf_heat_pump_suitability.pipeline.error_analysis.error_analysis_utils import *
+from asf_heat_pump_suitability.pipeline.reweight_epc import prepare_target
+from asf_heat_pump_suitability.getters.s3_getters import save_to_s3
+from asf_heat_pump_suitability import logger
+
+main_error_metrics = [
+    "rmse_all_cats",
+    "mae_all_cats",
+    "rmse_missing_cats",
+    "mae_missing_cats",
+]
+
+evaluation_feature_names = ["tenure", "property_type", "build_year", "nrooms"]
+
+
+def parse_arguments():
+
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "--reweighted_dir",
+        help="S3 URI to weighted EPC dataset",
+        type=str,
+        default="s3://asf-heat-pump-suitability/outputs/2023_Q2_EPC_enhanced_weights.parquet",
+    )
+
+    parser.add_argument(
+        "--save_output",
+        help="S3 path to save evaluation results to",
+        type=str,
+        default=None,
+    )
+
+    parser.add_argument(
+        "--sample",
+        help="Whether to test this script on a sample or not",
+        default=False,
+        action="store_true",
+    )
+    args = parser.parse_args()
+
+    return args
+
+
+def get_errors_for_lsoa(
+    feature_name: str,
+    lsoa_code: str,
+    target_features: dict,
+    target_marginals: dict,
+    epc_subset: pl.DataFrame,
+) -> dict:
+    """
+    Get errors between target and unweighted, and target and weighted, for one feature and one LSOA.
+
+    Args:
+                    feature_name (str): The feature name to calculate errors for
+                    lsoa_code (str): The LSOA
+                    target_features (dict): A nested dictionary of the counts for all property features and LSOAs
+                    target_marginals (dict): A nested dictionary of the proportions for all property features and LSOAs
+                    epc_subset (pl.DataFrame): EPC dataset for this LSOA
+
+    Returns:
+                    dict: A dictionary containing various metrics for both the unweighted and weighted EPC data compared to
+                                    the target dataset for this feature.
+
+    """
+
+    target_counts = target_features[feature_name].get(lsoa_code)
+    target_proportions = target_marginals[feature_name].get(lsoa_code)
+
+    if target_counts:
+        orig_counts_list = epc_subset[feature_name].value_counts().to_dict()
```

Comment:
```suggestion
        orig_counts_dict = epc_subset[feature_name].value_counts().to_dict()
```
I got confused by the name on line 95 :D